maas/data_source_maas_boot_source_selection.go

```diff
@@ -0,0 +1,118 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceMaasBootSourceSelection() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMaasBootSourceSelectionRead,
+		Description: "Provides a resource to fetch a MAAS boot source selection.",
+
+		Schema: map[string]*schema.Schema{
+			"arches": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The architecture list for this resource",
+			},
+			"boot_source_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The BootSource this resource is associated with",
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The label lists for this resource",
+			},
+			"os": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Operating system for this resource",
+			},
+			"release": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The specific release of the Operating system for this resource",
+			},
+			"subarches": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The list of subarches for this resource",
+			},
+		},
+	}
+}
+
+func dataSourceMaasBootSourceSelectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ClientConfig).Client
+
+	bootsource, err := getBootSource(client)
```

Comment:
please use the ID provided by the user at the `boot_source` field given that it is renamed